types/convert.go

```diff
@@ -474,16 +477,39 @@ func floatStrToIntStr(sc *stmtctx.StatementContext, validFloat string, oriStr st
 	} else {
 		// convert scientific notation decimal number
 		extraZeroCount := intCnt - len(digits)
-		if extraZeroCount > 20 {
-			// Append overflow warning and return to avoid allocating too much memory.
-			sc.AppendWarning(ErrOverflow.GenWithStackByArgs("BIGINT", oriStr))
-			return validFloat[:eIdx], nil
-		}
 		intStr = string(digits) + strings.Repeat("0", extraZeroCount)
 	}
+	if isOverflowInt64(intStr) {
```

Comment:
I don't think we should check the overflow here because the duty of this function is to extract the valid int string prefix from a float string.